### DISCUSSION TITLE
Flower keeps the stale data of killed workers in metrics port when eks spins up the worker

### DISCUSSION THREAD

**Discussion by ajamadar-mdsol** [mher/flower#1414] [Q&A]
Hello,

I'm experiencing an issue with stale worker data in Flower metrics. I've set up Flower to expose its metric port, allowing Prometheus to scrape the metrics. My Celery workers are deployed on EKS, and when a new deployment is triggered, the existing workers are terminated and replaced with new ones. This results in a situation where there are 12 workers shown as online in the Flower UI, even though only 6 new workers are actually running. if i restart the flower then it purges the old workers.

I'm using the purge_offline_worker configuration to try to remove the stale worker data from the UI, but it's not having the desired effect. The stale workers are not being purged from the metrics, and their status is not being updated to offline. As a result, I'm seeing duplicate worker entries in the Flower UI, which is causing confusion and making it difficult to manage my workers.

Can anyone suggest a solution or a configuration that I can use to get rid of stale worker data in Flower metrics? I'd appreciate any help or guidance on this issue.

**Comment by manviii27** [comment]
You can try setting the max_workers config for flower to 6 (assuming your online workers will always be 6)
